### ISSUE TITLE
Memory Inspector display enhancements

### ISSUE THREAD

**Issue by jreineckearm** [eclipse-cdt-cloud/vscode-memory-inspector#48]
**Description**

As discussed during the last CDT Cloud meeting, I'd like to share some thoughts on potential display enhancements which could make the Memory Inspector even more powerful than it is already today.
* Configure word size in bytes. As I understand this is currently fixed to 1-byte words.
* "autofit" option for "Groups Per Row" to make most use of the current window/data column size.
* Space separator or similar between data words. This would make it easier to read words especially in combination with endianess.
* Display of different data representations (decimal, octal, binary, float, double, signed vs unsigned, etc)
* An extension point for more CPU architecture specific representations, e.g. special floating or fixed point formats.
* Pad address values with leading 0's according to architecture, e.g. always 8-digit hex values for 32-bit architectures.
* Selectable word endianess, for example if viewing a memory buffer for a peripheral which operates with a different endianess than the CPU.
* Option to show/hide the base prefix of memory values (O, 0x, 0b, etc)
* Configuration of physical access size in bytes. (Understood that currently not supported by MS-DAP)
* Show most common formats for a selected word in a separate area

**Additional information**

**Comment by colin-grant-work** [comment]
At Ericsson, we're currently working on reintroducing the hover system that (among other things) displayed values in different formats, so we'd be happy to take on 
> Display of different data representations (decimal, octal, binary, float, double, signed vs unsigned, etc)
@gbodeen, re: some possible format additions.

**Comment by colin-grant-work** [comment]
> Configure word size in bytes. As I understand this is currently fixed to 1-byte words.

It would be nice for this one if it were included as an extension point for debuggers that could answer a query about word size for a given session.

**Comment by jreineckearm** [comment]
> At Ericsson, we're currently working on reintroducing the hover system that (among other things) displayed values in different formats, so we'd be happy to take on

That would be awesome, thank you @colin-grant-work ! Would this agree with your plans, @planger ?

**Comment by jreineckearm** [comment]
> It would be nice for this one if it were included as an extension point for debuggers that could answer a query about word size for a given session.

I fully agree. Ideally, this knowledge would be transported via the MS-DAP protocol. Need to investigate which would be the best type or response. It certainly will have its charm also for other windows and debug functionality.

A short-term/intermediate alternative could be to add it as another setting. That could later pick up knowledge from the debugger through MS-DAP with an `auto` setting.
Finding a good default for this setting may become a challenge. Maybe it could be steered by an agreed debug launch configuration item like it's used for description files for the Peripheral Inspector extension?

**Comment by planger** [comment]
> > At Ericsson, we're currently working on reintroducing the hover system that (among other things) displayed values in different formats, so we'd be happy to take on
> 
> That would be awesome, thank you @colin-grant-work ! Would this agree with your plans, @planger ?

Sure, that sounds great! I'll comment on https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/issues/66 with our initial thoughts to keep the discussion in the right place.
